File: my_first_package/src/param_node.cpp
#include <chrono>
#include <functional>
#include <string>
#include <rclcpp/rclcpp.hpp>

using namespace std::chrono_literals;

class MinimalParam : public rclcpp::Node {
public:
  MinimalParam() : Node("minimal_param_node") {
    // 1. 声明参数
    this->declare_parameter("my_config_param", "new_default_value");

    // 2. 创建定时器，每秒执行一次回调
    timer_ = this->create_wall_timer(
      1000ms, std::bind(&MinimalParam::timer_callback, this));
  }

  void timer_callback() {
    // 3. 实时获取参数值
    std::string my_param = this->get_parameter("my_config_param").as_string();
    RCLCPP_INFO(this->get_logger(), "C++ Node Value: %s", my_param.c_str());
  }

private:
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char ** argv) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MinimalParam>());
  rclcpp::shutdown();
  return 0;
}

File: my_first_package/my_first_package/param_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MinimalParam(Node):
    def __init__(self):
        super().__init__('minimal_param_node')
        # 声明参数
        self.declare_parameter('my_config_param', 'default_value')
        
        # 创建定时器，每 1.0 秒调用一次下面的 self.timer_callback
        self.timer = self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        # 每次执行都要实时 get 一下参数值
        my_param = self.get_parameter('my_config_param').get_parameter_value().string_value
        
        self.get_logger().info('Value: %s' % my_param)

def main():
    rclpy.init()
    node = MinimalParam()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()
